packages/host/app/services/realm.ts

```diff
@@ -55,11 +56,19 @@ class RealmResource {
   @service private declare messageService: MessageService;
 
   @tracked info: EnhancedRealmInfo | undefined;
+  @tracked private realmPermissions: RealmPermissions | null | undefined;
 
   @tracked
   private auth: AuthStatus = { type: 'anonymous' };
   private subscription: { unsubscribe: () => void } | undefined;
 
+  // Hassan: in general i'm questioning the usefulness of using Tasks in this
+  // class. We seem to be following the pattern of await-ing all the tasks on
+  // the outside of the Task.perform(). When we do this it actually casts the
+  // Task.perform() into a normal non-cancellable promise. Probably we can get
+  // rid of all these tasks and just use normal promises as I don't think the
+  // tasks are buying us anything.
+
```

Comment:
It seems that in the logout method, we cancel running tasks. That would not be possible if we were not using tasks.

It's true that awaiting a task instance outside a task means you are dealing with a regular promise, but that is not the only way we interact with tasks. Furthermore task instances being awaited that way can still be canceled, have concurrency config (e.g. `dropTask`), and have derived state (not sure if we are using that in this case).

Reply:
So the cancelling happens behind the awaited perform boundary. But I think we don’t actually have a problem with that because during logout we coincidentally don’t happen to be performing any running tasks. If a task was running during logout and we did the cancel I’m pretty sure we’d get the cancel task exception because the running task actually got cast to a promise because of how we await these tasks in our API. 

In fact I did actually run into a form of this in a different PR that I had to work around by caching the promise because dropTask wasn’t able to allow subsequent performs during a running task to hook into the async for the already running task.